### ISSUE TITLE
Deployment of the MACE Model Without Using ASE

### ISSUE THREAD

**Issue by Hsuchein** [ACEsuit/mace#511]
First of all, I would like to express my gratitude for your efforts in developing the excellent MACE architecture.

Recently, I have been considering whether it is possible to package MACE into a .pt file using Torch's JIT technology and then deploy it using C++. This could potentially improve the inference speed of MACE and seamlessly integrate it with my C++ code, which is exactly what I am aiming for.

However, after going through the tutorial, it seems that MACE can only function as a calculator that is invoked by the atoms object in ASE for computation. My project does not require ASE, and the presence of ASE creates issues when packaging the model.

Ideally, I would like to achieve something like outputs = model.forward({species, coordinates}), where MACE acts as a standalone model. I would only need to input the atomic types (species) and atomic coordinates (coordinates) to compute the energy and potential.

Below is the code I have tried, but it resulted in an error. Could you please advise if MACE can be used independently of ASE, saved as a .pt model, and deployed with the Libtorch library to handle inputs and output energy and force?

Thank you for your attention to this matter.

`import torch
from mace.calculators import mace_off
from ase import Atoms

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

species = ['C', 'C']
coordinates = [
    [0.0, 0.0, 0.0],
    [1.0, 0.0, 0.0]
]

class Wrapper(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.calc = mace_off(model="medium", device='cuda')

    def forward(self, species, coordinates):
        
        atoms = Atoms(symbols=species, positions=coordinates)
        atoms.set_calculator(self.calc)
        energy = atoms.get_potential_energy()
        forces = atoms.get_forces()
        
        return energy, forces

wrapped_model = Wrapper()

energy, forces = wrapped_model(species, coordinates)

print(f"Energy: {energy}")
print(f"Forces: {forces}")

traced_model = torch.jit.script(wrapped_model)

traced_model.save("/home/wayne/data/mace/mace_medium.pt")`

**Comment by CheukHinHoJerry** [comment]
ASE calculator is not a must as long as you provide a dictionary that is correct for call `model.forward`. Take a look at 
`https://github.com/ACEsuit/mace/blob/main/mace/data/atomic_data.py` and how it is used in testing scripts etc.

**Comment by bernstei** [comment]
Isn't the neighbor list calculated in python, which would therefore make it impossible to run in pure C++ without some significant supporting code?

**Comment by CheukHinHoJerry** [comment]
Yes. I would imagine you will have to write an interface for it. I am not familiar with nlist in C++ but it is at least doable with reasonable amount of time I think.

**Comment by ilyes319** [comment]
Hey @CheukHinHoJerry,

Yes it is totally possible to run MACE in C++ using torchscript. This exactly how we made the lammps interface, see here: https://github.com/ACEsuit/lammps/blob/mace/src/ML-MACE/pair_mace.cpp#L308.
As @CheukHinHoJerry you will just need to provide the right input dictionary. To construct the NL you should be able to use the C++ matscipy NN.

**Comment by Hsuchein** [comment]
I guess you'r right. I find a code in `eval_configs.py`, which use `output = model(batch.to_dict(), compute_stress=args.compute_stress)` to obtain the force and energy. Is that means what I should do is to provide the model with right dictionary, does it make sense?

Furthermore, I am not quite sure about nblist you guys mentioned, should I provide the model with nblist?
